extensions/sql-database-projects/src/models/deploy/deployService.ts

```diff
@@ -0,0 +1,403 @@
+		const getAzdataApi = await utils.getAzdataApi();
+		let connection = await this.retry(
+			constants.connectingToSqlServerOnDockerMessage,
+			async () => {
+				return await this.connectToDatabase(profile, savePassword, database);
+			},
+			this.validateConnection,
+			this.formatConnectionResult,
+			5, timeoutInSeconds);
+
+		if (connection) {
+			const connectionResult = <ConnectionResult>connection;
+			if (getAzdataApi) {
+				return await getAzdataApi.connection.getUriForConnection(connectionResult.connectionId);
+			} else {
+				return <string>connection;
+			}
+		}
+
+		return undefined;
+	}
+
+	private async executeTask<T>(taskName: string, task: () => Promise<T>): Promise<T> {
+		const getAzdataApi = await utils.getAzdataApi();
+		if (getAzdataApi) {
+			return new Promise<T>((resolve, reject) => {
+				let msgTaskName = taskName;
+				getAzdataApi!.tasks.startBackgroundOperation({
+					displayName: msgTaskName,
+					description: msgTaskName,
+					isCancelable: false,
+					operation: async op => {
+						try {
+							let result: T = await task();
+
+							op.updateStatus(getAzdataApi!.TaskStatus.Succeeded);
+							resolve(result);
+						} catch (error) {
+							let errorMsg = constants.taskFailedError(taskName, error ? error.message : '');
+							op.updateStatus(getAzdataApi!.TaskStatus.Failed, errorMsg);
+							reject(errorMsg);
+						}
+					}
+				});
+			});
+		} else {
+			return await task();
+		}
+	}
+
+	private async retry<T>(
+		name: string,
+		attempt: () => Promise<T>,
+		verify: (result: T) => Promise<ValidationResult>,
+		formatResult: (result: T) => Promise<string>,
+		numberOfAttempts: number = 10,
+		waitInSeconds: number = 2): Promise<T | undefined> {
+		for (let count = 0; count < numberOfAttempts; count++) {
+			this.logToOutput(constants.retryWaitMessage(waitInSeconds, name));
+			await new Promise(c => setTimeout(c, waitInSeconds * 1000));
+			this.logToOutput(constants.retryRunMessage(count, numberOfAttempts, name));
+
+			try {
+				let result = await attempt();
+				const validationResult = await verify(result);
+				const formattedResult = await formatResult(result);
+				if (validationResult.validated) {
+					this.logToOutput(constants.retrySucceedMessage(name, formattedResult));
+					return result;
+				} else {
+					this.logToOutput(constants.retryFailedMessage(name, formattedResult, validationResult.errorMessage));
+				}
+
+			} catch (err) {
+				this.logToOutput(constants.retryMessage(name, err));
+			}
+		}
+
+		return undefined;
+	}
+
+	private logToOutput(message: string): void {
+		if (this._outputChannel) {
+			this._outputChannel.appendLine(message);
+		}
+	}
+
+	private async createCommands(mssqlFolderPath: string, commandsFolderPath: string, dockerFilePath: string, startFilePath: string, imageLabel: string): Promise<void> {
```

Comment:
Comments again. What commands? It also looks like it should be `createCommandFiles`